codex-rs/core/src/codex.rs

```diff
@@ -1474,6 +1478,101 @@ async fn submission_loop(
+
+                    // Build a shell invocation in the user's default shell.
+                    let shell_invocation = sess_clone
+                        .user_shell
+                        // Why we pass a ["bash", "-lc", <script>] sentinel instead of the raw command:
+                        // - The shell adapter (core/src/shell.rs) first calls `strip_bash_lc`. When it sees this
+                        //   exact shape it extracts <script> and then builds the correct argv for the user shell
+                        //   (e.g., `/bin/zsh -lc "source ~/.zshrc && (<script>)"`).
+                        // - If we pass the whole command as a single string (e.g., ["cat Cargo.toml | wc -l"]) the
+                        //   adapter may quote it when joining/embedding, and shells can treat the entire value as a
+                        //   single program name or a single quoted token.
+                        .format_default_shell_invocation(vec![
+                            "bash".to_string(),
+                            "-lc".to_string(),
+                            command.clone(),
+                        ])
+                        .unwrap_or_else(|| vec![command.clone()]);
+
+                    let params = ExecParams {
+                        command: shell_invocation.clone(),
+                        cwd: turn_context.cwd.clone(),
+                        timeout_ms: None,
+                        env: create_env(&turn_context.shell_environment_policy),
+                        with_escalated_permissions: None,
+                        justification: None,
+                    };
+
+                    // Use a fresh diff tracker (no patch application expected for ! commands).
+                    let mut turn_diff_tracker = TurnDiffTracker::new();
+                    // Initiated by user, not by the model. Hence, we generate a new call_id.
+                    let call_id = format!("call_{}", Uuid::new_v4());
+
+                    let exec_ctx = ExecCommandContext {
+                        sub_id: sub_id.clone(),
+                        call_id: call_id.clone(),
+                        command_for_display: shell_invocation,
+                        cwd: params.cwd.clone(),
+                        apply_patch: None,
+                        user_initiated_shell_command: true,
+                    };
+
+                    // Run without sandboxing or approval — this is a user-initiated command.
+                    // Output is not captured as it's sent to the TUI inside `run_exec_with_events`.
+                    let _ = sess_clone
+                        .run_exec_with_events(
+                            &mut turn_diff_tracker,
+                            exec_ctx,
+                            ExecInvokeArgs {
+                                params,
+                                sandbox_type: SandboxType::None,
+                                sandbox_policy: &turn_context.sandbox_policy,
```

Comment:
Since this is user-initiated, should it be run without a sandbox?

```suggestion
                                sandbox_policy: SandboxPolicy::DangerFullAccess,
```